evap/evaluation/views.py

```diff
@@ -80,10 +82,45 @@ def index(request):
             return redirect('results:index')
 
 
-def login_key_redirect(request, key):
-    response = redirect('evaluation:index')
-    response['Location'] += '?loginkey=' + str(key)
-    return response
+def login_key_authentication(request, key):
+    # AuthenticationMiddleware is required so that request.user exists.
+    if not hasattr(request, 'user'):
+        raise ImproperlyConfigured(
+            "This view requires the"
+            " authentication middleware to be installed.  Edit your"
+            " MIDDLEWARE_CLASSES setting to insert"
+            " 'django.contrib.auth.middleware.AuthenticationMiddleware'")
```

Comment:
i think that part has outlived its usefulness. if that middleware is missing, i'm sure there's a massive bunch of stuff failing loudly, i don't know why of all things we should test for this particular middleware to exist.